crates/ra_hir_ty/src/display.rs

```diff
@@ -322,7 +320,7 @@ impl HirDisplay for Ty {
 }
 
 fn write_bounds_like_dyn_trait(
-    predicates: &[GenericPredicate],
+    predicates: impl IntoIterator<Item = impl Borrow<GenericPredicate>>,
```

Comment:
That feel, when one invocation passes `Item = GenericPredicate`, but the other is `Item = &GenericPredicate`. Not sure if my solution is overengineered, I can leave intermediate `.collect::<Vec<_>>()` if you wish

Reply:
I'd rather keep this function non-generic, it's rather large and def not hot, so we should avoid monomorphisation.

Reply:
I would think it's hot since inlay hints are recomputed on each character typed, but if you insist

Reply:
Anything related to presentation is not hot (or can be made not hot), as we present only few things to the user. 

Ie, `O(characters on the scren)` things are cold, `O(code in the project)` things ar hot.

Reply:
Done